src/Microsoft.AspNet.Mvc.ViewFeatures/ViewDataDictionary.cs

```diff
@@ -186,7 +186,9 @@ protected ViewDataDictionary([NotNull] ViewDataDictionary source, object model,
         {
             // This is the core constructor called when Model is known.
             var modelType = GetModelType(model);
-            if (modelType == source.ModelMetadata.ModelType && model == source.ModelExplorer.Model)
+            var metadataModelType =
+                Nullable.GetUnderlyingType(source.ModelMetadata.ModelType) ?? source.ModelMetadata.ModelType;
```

Comment:
I like the idea but suggest doing this (and a bit more) as a follow-up. Filed #2992